src/tree/map.rs

```diff
@@ -765,8 +755,38 @@ impl<K, V, C> TreeMap<K, V, C> where C: Compare<K> {
     /// assert_eq!(map.get(&6), Some(&"changed"));
     /// assert_eq!(map.get(&8), Some(&"changed"));
     /// ```
-    pub fn lower_bound_mut<Q: ?Sized>(&mut self, k: &Q) -> IterMut<K, V> where C: Compare<Q, K> {
-        bound_setup!(self.iter_mut_for_traversal(), k, true)
+    pub fn lower_bound_mut<Q: ?Sized>(&mut self, key: &Q) -> BoundIterMut<K, V>
+        where C: Compare<Q, K> {
+
+        BoundIterMut {
+            iter: node::Iter::bounded(&self.root, &self.cmp, key, true, self.len)
+        }
+    }
+
+    /// Returns a lazy iterator to the first key-value pair whose key is greater than `k`
```

Comment:
Yeah, I haven't updated the docs yet. I just left them as they were, but they definitely need improvement and clarification.